application/src/main/java/org/togetherjava/tjbot/commands/moderation/UnbanCommand.java

```diff
@@ -0,0 +1,114 @@
+ * currently not banned.
+ */
+public final class UnbanCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(UnbanCommand.class);
+    private static final String TARGET_OPTION = "user";
+    private static final String REASON_OPTION = "reason";
+    private static final String COMMAND_NAME = "unban";
+    private static final String ACTION_VERB = "unban";
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Constructs an instance.
+     */
+    public UnbanCommand() {
+        super(COMMAND_NAME, "Unbans the given user from the server", SlashCommandVisibility.GUILD);
+
+        getData()
+            .addOption(OptionType.USER, TARGET_OPTION, "The banned user who you want to unban",
+                    true)
+            .addOption(OptionType.STRING, REASON_OPTION, "Why the user should be unbanned", true);
+
+        hasRequiredRole = Pattern.compile(Config.getInstance().getHeavyModerationRolePattern())
+            .asMatchPredicate();
+    }
+
+    private static void unban(@NotNull User target, @NotNull Member author, @NotNull String reason,
+            @NotNull Guild guild, @NotNull Interaction event) {
+        String targetTag = target.getAsTag();
+
+        guild.unban(target).reason(reason).queue(result -> {
+            MessageEmbed message = ModerationUtils.createActionResponse(author.getUser(),
+                    ModerationUtils.Action.UNBAN, target, null, reason);
+            event.replyEmbeds(message).queue();
+
+            logger.info("'{}' ({}) unbanned the user '{}' ({}) from guild '{}' for reason '{}'.",
+                    author.getUser().getAsTag(), author.getId(), targetTag, target.getId(),
+                    guild.getName(), reason);
+        }, unbanFailure -> {
+            if (unbanFailure instanceof ErrorResponseException errorResponseException) {
+                if (errorResponseException.getErrorResponse() == ErrorResponse.UNKNOWN_USER) {
+                    event.reply("The specified user does not exist.").setEphemeral(true).queue();
+                    logger.debug("Unable to unban the user '{}' because they do not exist.",
+                            targetTag);
+                    return;
+                }
+                if (errorResponseException.getErrorResponse() == ErrorResponse.UNKNOWN_BAN) {
+                    event.reply("The specified user is not banned.").setEphemeral(true).queue();
+                    logger.debug("Unable to unban the user '{}' because they are not banned.",
+                            targetTag);
+                    return;
+                }
+            }
+            event.reply("Sorry, but something went wrong.").setEphemeral(true).queue();
+            logger.warn("Something unexpected went wrong while trying to unban the user '{}'.",
+                    targetTag, unbanFailure);
+        });
```

Comment:
Create a method for all this failure handling?

Reply:
This code is not part of this PR. Closing.